assets/src/dashboard/components/cardGridItem/cardPreview.js

```diff
@@ -80,47 +74,57 @@ const CtaContainer = styled.div`
   margin: auto auto 25%;
 `;
 
-const PlayArrowIcon = styled(PlayArrowSvg).attrs({ width: 11, height: 14 })`
-  margin-right: 9px;
-`;
+const BottomActionButton = styled(Button)``;
 
-const EditButton = styled(Button).attrs({ onClick: () => {} })``;
+const getActionAttributes = (targetAction) =>
+  typeof targetAction === 'string'
+    ? { forwardedAs: 'a', href: targetAction, onClick: () => {} }
+    : { onClick: targetAction };
 
-// TODO modify to handle other types of grid items, not just own stories
-const CardPreviewContainer = ({ editUrl, onPreviewClick, children }) => {
-  const displayEditControls = onPreviewClick || editUrl;
+const CardPreviewContainer = ({ centerAction, bottomAction, children }) => {
   const { pageSize } = usePagePreviewSize();
+
   return (
     <>
       <PreviewPane cardSize={pageSize}>{children}</PreviewPane>
-      {displayEditControls && (
-        <EditControls cardSize={pageSize}>
-          {onPreviewClick && (
-            <PreviewContainer>
-              <PreviewButton
-                type={BUTTON_TYPES.SECONDARY}
-                onClick={onPreviewClick}
-              >
-                <PlayArrowIcon />
-                {__('Preview', 'web-stories')}
-              </PreviewButton>
-            </PreviewContainer>
-          )}
-          <CtaContainer>
-            <EditButton forwardedAs="a" href={editUrl}>
-              {__('Open in editor', 'web-stories')}
-            </EditButton>
-          </CtaContainer>
-        </EditControls>
-      )}
+      <EditControls cardSize={pageSize}>
+        {centerAction && (
+          <CenterActionContainer>
+            <CenterActionButton
+              type={BUTTON_TYPES.SECONDARY}
+              {...getActionAttributes(centerAction.targetAction)}
+            >
+              {centerAction.label}
+            </CenterActionButton>
+          </CenterActionContainer>
+        )}
+        <CtaContainer>
+          <BottomActionButton
+            {...getActionAttributes(bottomAction.targetAction)}
+          >
+            {bottomAction.label}
+          </BottomActionButton>
+        </CtaContainer>
+      </EditControls>
     </>
   );
 };
 
+export const ActionLabel = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.node,
+]);
+
+export const ActionButtonPropType = PropTypes.shape({
+  targetAction: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
+    .isRequired,
+  label: ActionLabel.isRequired,
+});
+
 CardPreviewContainer.propTypes = {
   children: PropTypes.node.isRequired,
-  editUrl: PropTypes.string.isRequired,
-  onPreviewClick: PropTypes.func,
+  centerAction: ActionButtonPropType,
+  bottomAction: ActionButtonPropType.isRequired,
```

Comment:
I decided to settle on `center` and `bottom` actions (and we can later add in `top` action for bookmarks)... I figured it was generic/flexible/understandable enough to work, but feel free to push back on this pattern if you see problems with it :D